Project.toml

```diff
@@ -4,10 +4,12 @@ authors = ["Nathan Daly <nhdaly@gmail.com>", "Dana Wilson <odioustoad@gmail.com>
 version = "1.0.0"
 
 [deps]
+FlameGraphs = "08572546-2f56-4bcf-ba4e-bab62c3a3f89"
 HTTP = "cd3eb016-35fb-5094-929b-558a96fad6f3"
 PProf = "e4faabce-9ead-11e9-39d9-4379958e3056"
 Profile = "9abbd945-dff8-562f-b5e8-e1ebf5ef1b79"
 Serialization = "9e88b42a-f829-5b0c-bbe9-9e923198166b"
+SnoopCompile = "aa65fe97-06da-5843-b5b1-d5d13cad87d2"
```

Comment:
🤔 I remember that we wanted to avoid having SnoopCompile in the dependencies for our server, because it adds a bunch of expensive deps for Cthulhu and FlameGraphs and stuff...

I'm thinking maybe what we could do is just directly return the InferenceTimingNodes, and then have the _client_ do the conversion to PProf? You could send the profiles directly in their original form, via `Serialization.serialize()`, and then reconstruct them into pprof profiles on the client side? I think it would be okay to add these deps to our benchmarking package, rather than to our server.

Reply:
🤔 but actually, looking at this dependencies list, this doesn't seem sooo bad:
```
(ProfileEndpoints) pkg> add SnoopCompile FlameGraphs
   Resolving package versions...
    Updating `~/.julia/dev/ProfileEndpoints/Project.toml`
⌅ [08572546] + FlameGraphs v0.2.10
  [aa65fe97] + SnoopCompile v2.9.6
    Updating `~/.julia/dev/ProfileEndpoints/Manifest.toml`
  [da1fd8a2] + CodeTracking v1.2.0
  [f68482b8] + Cthulhu v2.7.6
  [1eca21be] + FoldingTrees v1.2.1
  [c3a54625] + JET v0.6.21
  [aa1ae85d] + JuliaInterpreter v0.9.20
  [6f1432cf] + LoweredCodeUtils v2.3.0
  [1914dd2f] + MacroTools v0.5.10
  [295af30f] + Revise v3.5.0
  [aa65fe97] + SnoopCompile v2.9.6
  [69024149] + StringEncodings v0.3.5
  [ddb6d928] + YAML v0.4.8
```

🤔 Do you remember why we wanted to avoid SnoopCompile as a dependency? 🤔

Reply:
Aha! I found the culprit -- it turned out it was actually an unrelated change in your PR, coming from PProf!
https://github.com/RelationalAI/raicode/pull/10713/files#diff-6d1205e03b7a43c01ed2ac8b8c83eb58c6f3a3b1360769e18644245abee8c565

It's this issue, here:
https://github.com/JuliaPerf/PProf.jl/issues/72, which we're now tracking here as well: https://relationalai.atlassian.net/browse/RAI-8826.

So i think this is actually maybe fine? 👍 
It'd still be nice to not have to depend on Revise, JET, and Cthulhu at runtime, though, if we could help it. So it might still be worth pulling this out and doing this work client-side.

Reply:
Okay drats, i tried this locally, but it actually doesn't work, since the inference profile directly references all the methods that were profiled, and those methods won't exist on the client. For example:
```julia
julia> deserialize("/Users/nathandaly/Downloads/inference_profile.jlserialized")
ERROR: KeyError: key HTTP [cd3eb016-35fb-5094-929b-558a96fad6f3] not found
```
So either we'd need to convert all those method references to a string, and build a new profile object (flamegraph), or we just do what you have here and import SnoopCompile to do that for us. 👍 

So I guess I'd say we leave this as-is, for now. 👍

Reply:
PR for the above, here: https://github.com/JuliaPerf/ProfileEndpoints.jl/pull/17, which i immediately closed.